docs/adr/0013-react-toolchain-for-settings-redesign.md

```diff
@@ -0,0 +1,118 @@
+
+As there are a multitude of ways a React App can be set up and configured, this decision has introduced the question, "What [toolchain](https://en.wikipedia.org/wiki/Toolchain) should we set up a new React App with?". This ADR serves to answer that question by going over the various approaches we can take to set up a new React App and how it might integrate with our existing FxA ecosystem.
+
+It's important to note that we are not deciding the languages, libraries, or other tools that we will use within the React development environment, but rather, the system that will be used to develop the React App with.
+
+Finally, we are not making decisions about the App's production server here. We'll be using [fxa-content-server](https://github.com/mozilla/fxa/tree/master/packages/fxa-content-server/server)'s Express server for the production build.
+
+## Decision Drivers
+
+- **Familiarity** - unless there is an overwhelming reason, we don't really want to introduce a completely foreign build process for everyone to learn.
+- **Configurability** - if it is less configurable does it meet our needs; if it is more configurable is the time spent worth it?
+- **Maintainability** - how much work do we set up for ourselves if we're maintaining all upgrades and configuration?
+
+Additionally, our team has a desire to share React components and other files across projects. Both the Payments and Admin Panel web apps use React, and we would benefit from being able to develop components that could be imported into any of these React Apps.
+
+## Considered Options
+
+- Option A - Completely Custom Approach
+- Option B - Create React App
+  - Option B2 - Create React App and immediately eject
+  - Option B3 - Create React App and use Rescripts
+- Option C - Neutrino with React Preset
+- Option D - Other Considerations
+
+## Decision Outcome
+
+Chosen option: "B3 - Create React App and use Rescripts", because:
+
+- Our team has sufficient knowledge in working with CRA over the other options.
+- We don't need to re-invent the wheel. Our needs for the Settings project are not so great that it would be worth spending time creating a ground-up toolchain. To some extent this option allows us to do what we do best without needing to worry about the specifics of the development workflow.
+- Rescripts allows us to layer additional configuration overtop of the existing configuration without jeopardizing the integrity or trading away any of the powers of CRA. Specifically, we can use Rescripts to modify CRA to allow external React components to be imported, as demonstrated in [this repo](https://github.com/jodyheavener/rescripts-react-app) (technical notes [here](https://github.com/jodyheavener/rescripts-react-app/blob/master/.rescriptsrc.js)). This capability could be implemented in our other React Apps.
```

Comment:
🙌 Thanks for making that proof of concept. While we may end up modifying it, it's great to see something in action working to set us up to share React components in our CRAs across our monorepo with this decision.